### PR TITLE
[Models] Add PaliGemma-3B-224 vision-language model integration

### PR DESCRIPTION
**Pull request by LibbyStarr10**
- PaliGemmaForConditionalGeneration: SigLIP vision (27 layers) + Gemma text (18 layers)
- 1 monkey-patch for get_placeholder_mask (data-dependent assert)
- Single forward graph (VLM)

## Summary

Add PaliGemma-3B-224 as an official model under `models/paligemma/`, extending buddy-mlir VLM coverage.

**Architecture**: PaliGemmaForConditionalGeneration — SigLIP vision encoder (27 layers, hidden=1152) projects image features through a linear projector (1152→2048) into a Gemma text decoder (18 layers, hidden=2048, 8 heads, 1 KV head). 256 image tokens per 224×224 image.

**Integration approach**: Single forward graph. One monkey-patch on `get_placeholder_mask` removes a data-dependent `numel()` assert (same pattern as PaddleOCR). The model is otherwise a standard HuggingFace model.

**Files added** (8 files, 208 lines):
- `models/paligemma/CMakeLists.txt` — `buddy_add_model` build config
- `models/paligemma/import_model.py` — PyTorch → MLIR importer
- `models/paligemma/PaligemmaRunner.cpp` — inference runner
- `models/paligemma/include/buddy/runtime/models/PaligemmaRunner.h`
- `models/paligemma/specs/f32.json` — variant spec
- `models/paligemma/validate_accuracy.py` — logits comparison
- `models/CMakeLists.txt` — register with `BUDDY_BUILD_PALIGEMMA_MODEL` guard

## Verification

```bash
cd models/paligemma
python import_model.py --output-dir .
ls layer_partitioned/   # forward.mlir, subgraph0.mlir
ls -lh arg0.data        # ~11 GB (3B f32 params)

### CODE REVIEW

**Comment by jiaan-clone** [comment]
Please resolve the code conflicts

**Comment by jiaan-clone** [comment]
Please standardize the code format
